Script fixes

asreek opened the pull request:
This activity contains the following changes:
1. Added support in the c++ class generation script to support counter types as data members. Fixed a bug in state class template file to add unique headers for "next state" classes, in case the same next-state entry exists for multiple events.
Both these changes got reverted during security feature submission.
2. Removed duplicate entries from the stateMachineAppModel.json, ctxtManagerAppModel.json. 
3. Nwk Init detach request fix.

omecproject commented:
Can one of the admins verify this patch?

omecproject commented:
Can one of the admins verify this patch?

omecproject commented:
Can one of the admins verify this patch?

omecproject commented:
Can one of the admins verify this patch?

sindack commented:
ok to test

sindack commented:
test this please

sindack commented:
There may be a problem with the CORD Jenkins webhook, looking into it, apologies ahead of time for bot command comments

sindack commented:
retest this please

yoooou commented:
retest this please

sindack commented:
retest this please

sindack commented:
retest this please

asreek commented:
@badhrinathpa , @thakurajayL 
Hi Badhri, Ajay
Added the attach collision fixes to this activity, since it has json changes as well. Please review.

PFA, PCAPs.
[Attach Collision TCs.zip](https://github.com/omec-project/Nucleus/files/4957696/Attach.Collision.TCs.zip)


Also, updated the code generation output path to point to source directories to avoid copying around the generated files.

Thanks,
Anjana

badhrinathpa commented on src/mme-app/mmeStates/attachWfMbResp.cpp:
```diff
@@ -63,4 +63,13 @@ void AttachWfMbResp::initialize()
+                ActionTable actionTable;
+                actionTable.addAction(&ActionHandlers::handle_attach_request);
+                actionTable.addAction(&ActionHandlers::del_session_req);
+                actionTable.addAction(&ActionHandlers::send_s1_rel_cmd_to_ue);
```
doing S1 release might be of no use cause already Attach has come on new ENB S1apID.

badhrinathpa commented on src/mme-app/mmeStates/s1ReleaseWfReleaseAccessBearerResp.cpp:
```diff
@@ -64,4 +64,13 @@ void S1ReleaseWfReleaseAccessBearerResp::initialize()
+                ActionTable actionTable;
+                actionTable.addAction(&ActionHandlers::handle_attach_request);
+                actionTable.addAction(&ActionHandlers::del_session_req);
+                actionTable.addAction(&ActionHandlers::send_s1_rel_cmd_to_ue);
```
here also sending S1 release is not required I think. We can clean up locally.

asreek replied:
Since we have received the attach on a new s1, and from mme perspective, another procedure is still in progress..shouldn't mme release the current s1?

asreek replied:
Removed s1 release as discussed.

badhrinathpa replied:
compilation is failing after the change. @asreek

sindack commented:
retest this please

sindack commented:
retest this please

sindack commented:
retest this please

yoooou commented:
retest this please

thakurajayL commented:
@asreek and @badhrinathpa  can you discuss and resolve this conflict.

thakurajayL commented:
@badhrinathpa can you please review

yoooou commented:
retest this please

yoooou commented:
retest this please
